libs/vkd3d/resource.c

```diff
@@ -2821,19 +2821,43 @@ static bool vk_descriptor_type_is_buffer(VkDescriptorType type)
 static void d3d12_desc_update_bindless_descriptor(struct d3d12_desc *dst)
 {
     const struct vkd3d_vk_device_procs *vk_procs = &dst->heap->device->vk_procs;
+    VkWriteDescriptorSet vk_writes[2], *vk_write;
     union vkd3d_descriptor_info descriptor_info;
     unsigned int descriptor_index, set_index;
     VkDescriptorSet vk_descriptor_set;
-    VkWriteDescriptorSet vk_write;
+    uint32_t write_count = 0;
     bool is_buffer;
 
     descriptor_index = d3d12_desc_heap_offset(dst);
 
     /* update UAV counter address */
-    if (dst->magic == VKD3D_DESCRIPTOR_MAGIC_UAV && dst->heap->uav_counters.data)
+    if (dst->magic == VKD3D_DESCRIPTOR_MAGIC_UAV)
     {
-        dst->heap->uav_counters.data[descriptor_index] = dst->info.view
-                ? dst->info.view->vk_counter_address : 0;
+        set_index = d3d12_descriptor_heap_uav_counter_set_index();
+
+        if (dst->heap->uav_counters.data)
+        {
+            dst->heap->uav_counters.data[descriptor_index] = dst->info.view
+                    ? dst->info.view->vk_counter_address : 0;
+        }
+        else if ((vk_descriptor_set = dst->heap->vk_descriptor_sets[set_index]))
+        {
+            VkBufferView null_view = VK_NULL_HANDLE;
```

Comment:
Do we require robustness2 now?

Reply:
We always have a non-NULL view if we don't have robustness2.

Reply:
That said, I wanted to change this piece of code slightly.